src/serializer/ResidualHeapSerializer.js

```diff
@@ -886,7 +886,12 @@ export class ResidualHeapSerializer {
     remainingProperties: Map<string, PropertyBinding>
   ): void {
     const realm = this.realm;
-    let lenProperty = Get(realm, val, "length");
+    let lenProperty;
+    if (val.isLeakedObject()) {
+      lenProperty = this.realm.evaluateWithoutLeakLogic(() => Get(realm, val, "length"));
+    } else {
+      lenProperty = Get(realm, val, "length");
+    }
```

Comment:
I wonder if it would make more sense to wrap the whole serialization process into `evaluateWithoutLeakLogic` than to special-case the length access.

If my understanding is right by that point we don't really "run" user code so we don't have to worry about it "using" the leaked values. Maybe I'm wrong though.